src/model_config_utils.h

```diff
@@ -279,4 +279,24 @@ TRITONSERVER_DataType DataTypeToTriton(const inference::DataType dtype);
 /// \return The data type.
 inference::DataType TritonToDataType(const TRITONSERVER_DataType dtype);
 
+/// Check if the model configs may only differ in instance group.
+/// \param old_config The old model config.
+/// \param new_config The new model config.
+/// \return True if the model configs are the same without checking instance
+/// group. False if they are different without checking instance group.
+bool IsOnlyInstanceGroupMayDiffers(
```

Comment:
I think we can rename it to `DifferInNonInstanceGroupConfig`?

| non instance group config changed | instance group config changed | -> | Can update? |
| --- | --- | --- | --- |
| No | No | -> | Yes |
| No | Yes | -> | Yes |
| Yes | No | -> | No |
| Yes | Yes | -> | No |

From the table above, we do not care if instance group is altered or not (since it is excluded), the only thing matter is whether config other than instance group is changed or not.

Reply:
DifferInNonInstanceGroupConfig is good.

Reply:
Eventually, `EquivalentInNonInstanceGroupConfig(...)` is used, because it avoids double negating like `!DifferInNonInstanceGroupConfig(...)`.

Update commit: [Improve model config util naming](https://github.com/triton-inference-server/core/pull/189/commits/c01ea3ba8f037068b384c38b51074a1b266caf57)